src/utils/db.ts

```diff
@@ -0,0 +1,109 @@
+import { incentivizedArchaeologists } from "../data/seeds";
+import { initializeApp } from "firebase/app";
+import { getFirestore, addDoc, getDocs, collection, query, where, orderBy, setDoc, doc } from "firebase/firestore";
+
+const firebaseConfig = {
+  apiKey: process.env.FIREBASE_API_KEY,
+  projectId: process.env.FIREBASE_PROJECT_ID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
+
+interface DialAttempt {
+  address: string;
+  peerId: string;
+  timestampOfDial: number;
+  connectionStatus: boolean;
+}
+
+interface UptimeStats {
+  dialAttempts: number;
+  successes: number;
+  uptimeRatio: number;
+}
+
+export const updateIncentivizedArchaeologists = async () => {
+  try {
+    console.log("Updating incentivizes archaeologists...", incentivizedArchaeologists.length);
+
+    for await (const address of incentivizedArchaeologists) {
+      await setDoc(doc(db, `incentivized_archaeologists/${address}`), { address });
+    }
+
+    console.log("DONE!");
+  } catch (e) {
+    console.error("Error adding document: ", e);
+  }
+};
+
+export const saveDialResults = (attempts: DialAttempt[], timestampOfDial: number) => {
+  try {
+    attempts.map(async record => {
+      addDoc(collection(db, "dial_attempts"), record);
+    });
+
+    addDoc(collection(db, "dial_times"), { time: timestampOfDial });
+  } catch (e) {
+    console.error("Error adding document: ", e);
+  }
+};
+
+export const getOnlineNodes = async () => {
+  try {
+    const dialTimes = await getDocs(query(collection(db, "dial_times"), orderBy("time", "desc")));
+    const lastDialTime = dialTimes.docs[0].get("time");
```

Comment:
Might be able to use `limitToLast` -- https://firebase.google.com/docs/reference/node/firebase.database.Query#limittolast

From what I can tell, using a limit will lessen the number of `read`s performed (although this answer is pretty old) --
https://stackoverflow.com/questions/47146115/firestore-pricing-does-the-amount-of-documents-in-a-collection-matters

Reply:
This is awesome, thanks!